MagneticField/ParametrizedEngine/test/testParabolicParametrizedMagneticField.cc

```diff
@@ -0,0 +1,127 @@
+#include "DataFormats/FWLite/interface/Record.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include "MagneticField/ParametrizedEngine/interface/alpaka/ParabolicParametrizedMagneticField.h"
+
+#include "FWCore/Utilities/interface/stringize.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+
+#include <iostream>
+#include <fstream>
+#include <Eigen/Core>
+
+using namespace edm;
+using namespace std;
+using namespace alpaka;
+using namespace ALPAKA_ACCELERATOR_NAMESPACE;
+using namespace ALPAKA_ACCELERATOR_NAMESPACE::magneticFieldParabolicPortable;
+using Vector3f = Eigen::Matrix<float, 3, 1>;
+
+struct MagneticFieldKernel {
+  template <typename TAcc, typename T>
+  ALPAKA_FN_ACC void operator()(
+      TAcc const& acc, T const* __restrict__ in, T* __restrict__ out, size_t size) const {
+    for (auto index : cms::alpakatools::uniform_elements(acc, size)) {
+      out[index][0] = 0;
+      out[index][1] = 0;
+      out[index][2] = magneticFieldAtPoint(in[index]);
+    }
+  }
+};
+
+int main() {
+  ifstream file;
+  edm::FileInPath mydata("MagneticField/Engine/data/Regression/referenceField_160812_RII_3_8T.bin");
+  file.open(mydata.fullPath().c_str(), ios::binary);
+
+  int count = 0;
+  float px, py, pz;
+  float bx, by, bz;
+  vector<Vector3f> points;
+  vector<GlobalVector> referenceB_vec;
+
+  int numberOfPoints = 100;
+  do {
+    if (!(file.read((char*)&px, sizeof(float)) && file.read((char*)&py, sizeof(float)) &&
+          file.read((char*)&pz, sizeof(float)) && file.read((char*)&bx, sizeof(float)) &&
+          file.read((char*)&by, sizeof(float)) && file.read((char*)&bz, sizeof(float))))
+      break;
+
+    const auto gp = GlobalPoint(px, py, pz);
+    if (gp.perp2() > Parameters::max_radius2 || fabs(gp.z()) > Parameters::max_z)
+      continue;
+
+    points.push_back(Vector3f(px, py, pz));
+    referenceB_vec.push_back(GlobalVector(bx, by, bz));
+    count++;
+  } while (count < numberOfPoints);
+
+  const size_t size = points.size();
+  // allocate the input and output host buffer in pinned memory accessible by the Platform devices
+  auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
+  auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
+  // fill the input buffers, and the output buffer with zeros
+  for (size_t i = 0; i < size; ++i) {
+    in_host[i] = points[i];
+    out_host[i] = {0.};
+  }
+
+  float resolution = 0.2;
+  float maxdelta = 0.;
+  int fail = 0;
+
+  // run the test on each device
+  for (auto const& device : cms::alpakatools::devices<Platform>()) {
+    auto queue = Queue(device);
+    // allocate input and output buffers on the device
+    auto points_dev = cms::alpakatools::make_device_buffer<float[]>(queue, size);
+    auto deviceB = cms::alpakatools::make_device_buffer<float[]>(queue, size);
```

Comment:
can you choose consistent names for `in_host`, `out_host`, `points_dev`, `deviceB` ?